### ISSUE TITLE
WARNING: another process is updating barman configuration files

### ISSUE THREAD

**Issue by otbutz** [EnterpriseDB/barman#921]
We recently started to receive sporadic cron triggered notification mails with the following content:

```
WARNING: another process is updating barman configuration files
```

Barman version: 3.10.0 (installed via apt.postgresql.org)

Barman crontab:

```
* * * * * barman [ -x /usr/bin/barman ] && /usr/bin/barman -q cron
```

A quick interactive test revealed that `barman cron` finishes quite fast without any warnings:

```
$ time barman cron
Starting WAL archiving for server postgres-01
Starting WAL archiving for server postgres-02

real    0m0.426s
user    0m0.340s
sys     0m0.064s
```

Since the warning is only triggered once in a while, I suspect that high disk IO is throttling a cron execution to the point where it collides with an upcoming one. Perhaps the possibility of this happening was overlooked in #886?

**Comment by gcalacoci** [comment]
Hi,

just to be sure I'm understanding correctly, are you suggesting that the message should be debug level or something else?

**Comment by otbutz** [comment]
Maybe the underlying check that triggers the warning could be improved? Or does cron actually update any configuration files?

**Comment by gcalacoci** [comment]
Cron does update config files if needed and is part of the new `model` and `config-update` features.

Thanks a lot for explaining what you meant and for your suggestions, may I ask how often this is happening for you?

**Comment by otbutz** [comment]
Multiple occurrences per day:

![grafik](https://github.com/EnterpriseDB/barman/assets/22315436/2b74ea9e-24d1-4af8-8977-f5e31493a1cc)

**Comment by mbalmer** [comment]
I see this happeing all the time, too

**Comment by martinmarques** [comment]
Given that these are just WARNING messages and they are almost certainly from overlapping cron executions, we are closing this issue

**Comment by martinmarques** [comment]
I have a better understanding of what could have happened here, and will look into this.

**Comment by mbalmer** [comment]
So you have an obvious bug, do nothing, and then simply close it? Wow. You really seem not to care about your software.Am 27.06.2026 um 18:21 schrieb Martín Marqués ***@***.***>:﻿
Closed #921 as not planned.

**Comment by mjw-pp** [comment]
The recent cases of this were almost certainly caused by a bug in the apt.postgresql.org packaging, which is set up to run `barman cron` both from cron and from a systemd timer.

Looks like it will be fixed in the next release (3.19.1-3):

https://salsa.debian.org/postgresql/barman/-/commit/1c6a78a91f9d485ba4fe2d2bf8a2dda984971d59

**Comment by otbutz** [comment]
That explanation checks out. There's an active barman systemd timer and also a cron entry on the affected host.

**Comment by martinmarques** [comment]
@mbalmer  this is something that @mnencia , the Debian maintainer, is handling. It's not a barman issue. Sorry for not explaining this while doing some housekeeping on the weekend.

**Comment by mbalmer** [comment]
Am 29.06.2026 um 11:03 schrieb Martín Marqués:
> *martinmarques* left a comment (EnterpriseDB/barman#921) 
> <https://github.com/EnterpriseDB/barman/issues/921#issuecomment-4830718292>
>
> @mbalmer <https://github.com/mbalmer> this is something that @mnencia 
> <https://github.com/mnencia> , the Debian maintainer, is handling. 
> It's not a barman issue. Sorry for not explaining this while doing 
> some housekeeping on the weekend.
>
Just FYI, I am running barman on AlmaLInux 9.

**Comment by mbalmer** [comment]
Am 29.06.2026 um 11:03 schrieb Martín Marqués:
> *martinmarques* left a comment (EnterpriseDB/barman#921) 
> <https://github.com/EnterpriseDB/barman/issues/921#issuecomment-4830718292>
>
> @mbalmer <https://github.com/mbalmer> this is something that @mnencia 
> <https://github.com/mnencia> , the Debian maintainer, is handling. 
> It's not a barman issue. Sorry for not explaining this while doing 
> some housekeeping on the weekend.
>
So I found the issue on my system.  No systemd timer, but two cronjobs.  
One under user barman, one in /etc/cront.d

tsk, tsk...

**Comment by martinmarques** [comment]
I don't understand the issue, or at least I don't understand why this is something that Barman has to fix. 

I mean, you are saying you have 2 cronjobs. What do they execute? What is the unexpected behavior you are seeing?

Having two cronjobs will only bring up WARNINGs (normal, and not something to be concerned of) and skip operations due to locks.

If you could explain what you are actually experiencing, and better debugging information, we'd see what we can do.

**Comment by mbalmer** [comment]
> *martinmarques* left a comment (EnterpriseDB/barman#921) 
> <https://github.com/EnterpriseDB/barman/issues/921#issuecomment-4831607159>
>
> I don't understand the issue, or at least I don't understand why this 
> is something that Barman has to fix.
>
This is nothing barman can fix.  The issue can be closed imho.
>
> I mean, you are saying you have 2 cronjobs. What do they execute? What 
> is the unexpected behavior you are seeing?
>
> Having two cronjobs will only bring up WARNINGs (normal, and not 
> something to be concerned of) and skip operations due to locks.
>
yes, but my case the cronjob caused an email to be sent, producing a lot 
of spam.

> If you could explain what you are actually experiencing, and better 
> debugging information, we'd see what we can do.
>
I had a cron job installed for ever.  And then the package maintainers 
decided to add one, too. So my system ended up having two crongjobs 
executing "barman cron".

This is definitely not the barman developers fault...Message ID: 
***@***.***>

**Comment by JamesPearsonEWS** [comment]
This was fixed for us after updating from 3.19.1-2.pgdg24.04+1 to 3.19.1-3.pgdg24.04+1, and other people have reported the same. 
https://github.com/EnterpriseDB/barman/issues/1169#issuecomment-4903181650

Hopefully this is fully resolved now!
